src/content/docs/en/guides/server-islands.mdx

```diff
@@ -0,0 +1,108 @@
+
+Your page will be rendered immediately with any specified [fallback content as a placeholder](#server-island-fallback-content). Then, the component's own contents are fetched on the client and displayed when available.
+
+Add the [`server:defer` directive](/en/reference/directives-reference/#server-directives) to any component on your page to turn it into its own island:
+
+```astro title="src/pages/index.astro" "server:defer"
+---
+import Avatar from '../components/Avatar.astro';
+---
+<Avatar server:defer />
+```
+
+These components can do [anything you normally would in an on-demand rendered page](/en/guides/on-demand-rendering/#on-demand-rendering-features), such as fetch content, and access cookies:
+
+```astro title="src/components/Avatar.astro"
+---
+import { getUserAvatar } from '../sessions';
+const userSession = Astro.cookies.get('session');
+const avatarURL = await getUserAvatar(userSession);
+---
+<img alt="User avatar" src={avatarURL} />
+```
+
+## Server island fallback content
+
+When using the `server:defer` attribute on a component to delay its rendering, you can "slot" in default loading content using the included named `"fallback"` slot.
+
+Your fallback content will be rendered along with the rest of the page initially on page load, and will be replaced with your component's own content when it is available.
+
+To add fallback content, add `slot="fallback"` on a child (other components or HTML elements) passed to your server island component:
+
+```astro
+---
+import Avatar from '../components/Avatar.astro';
+import GenericAvatar from '../components/GenericAvatar.astro';
+---
+<Avatar server:defer>
+  <GenericAvatar slot="fallback" />
+</Avatar>
+```
+
+This fallback content can be things like:
+
+- A generic avatar instead of the user's own.
+- Placeholder UI such as custom messages.
+- Loading indicators such as spinners.
+
+
+## How it works 
+
+Server island implementation happens mostly at build-time where component content is swapped out for a small script.
+
+Each of the islands marked with `server:defer` is split off into its own special route which the script fetches at run time. When Astro builds your site it will omit the component and inject a script in its place, and any content you’ve marked with `slot="fallback"`.
+
+When the page loads in the browser, these components will be requested to a special endpoint that renders them and returns the HTML. This means that users will see the most critical parts of the page instantly. Fallback content will be visible for a short amount of time before the dynamic islands are then loaded.
+
+Each island is loaded independently from the rest. This means a slower island won't delay the rest of your personalized content from being available.
+
+This rendering pattern was built to be portable. It does not depend on any server infrastructure so it will work with any host you have, from a Node.js server in a Docker container, to the serverless provider of your choice.
```

Comment:
```suggestion
This rendering pattern was built to be portable. It does not depend on any server infrastructure so it will work with any host you have, from a Node.js server in a Docker container to the serverless provider of your choice.
```